lib/connection/result/column.js

```diff
@@ -268,14 +276,89 @@ function convertRawBigInt(rawColumnValue) {
 function convertRawBoolean(rawColumnValue) {
   let ret;
 
-  if ((rawColumnValue === '1') || (rawColumnValue === 'TRUE')) {
+  if (rawColumnValue  === true || (rawColumnValue === '1') || (rawColumnValue === 'TRUE')) {
     ret = true;
-  } else if ((rawColumnValue === '0') || (rawColumnValue === 'FALSE')) {
+  } else if (rawColumnValue === false || (rawColumnValue === '0') || (rawColumnValue === 'FALSE')) {
     ret = false;
   }
 
   return ret;
 }
+/**
+ * Converts a raw column value of structured type object to javascript Object
+ *
+ * @param {String} rawColumnValue
+ *
+ * @returns {Object}
+ */
+function convertRawObject(rawColumnValue, column, context) {
+  const metadata =  context.metadataFields.reduce(function (map, obj) {
+    map[obj.name] = obj;
+    return map;
+  }, {});
+  const json = JSON.parse(rawColumnValue);
+  const result = {};
+
+  Object.keys(json).forEach(function (key) {
+    result[key] = mapStructuredTypeFieldValue(json[key], metadata[key], context, column);
+  });
+
+  return result;
+
+}
+
+function mapStructuredTypeFieldValue(columnValue, columnMetadata, context, column) {
+  let value;
+  switch (columnMetadata.type) {
+  case 'text':
+    value =  columnValue;
+    break;
+  case 'real':
+    value =  toValueFromNumber(convertRawNumber(columnValue));
+    break;
+  case 'fixed':
+    value =  toValueFromNumber(convertRawNumber(columnValue));
+    break;
+  case 'boolean':
+    value =  convertRawBoolean(columnValue);
+    break;
+  case 'timestamp_ltz':
+    context.format = context.statementParameters['TIMESTAMP_LTZ_OUTPUT_FORMAT'] || context.statementParameters['TIMESTAMP_OUTPUT_FORMAT'];
+    value = convertDateTimeString(columnValue, context.format, context.statementParameters['TIMEZONE'], column.getScale()).toSfDate();
+    break;
+  case 'timestamp_ntz':
+    context.format = context.statementParameters['TIMESTAMP_NTZ_OUTPUT_FORMAT'] || context.statementParameters['TIMESTAMP_OUTPUT_FORMAT'];
+    value = convertDateTimeString(columnValue, context.format, moment.tz.guess(), column.getScale()).toSfDate();
+    break;
+  case 'timestamp_tz':
+    context.format = context.statementParameters['TIMESTAMP_TZ_OUTPUT_FORMAT'] || context.statementParameters['TIMESTAMP_OUTPUT_FORMAT'];
+    value = convertDateTimeString(columnValue, context.format, context.statementParameters['TIMEZONE'], column.getScale()).toSfDate();
```

Comment:
Why session timezone? It should be a value from timestamp itself?

Reply:
The timezone is used later to create correct string representation in method toJSON